lib/o_data/active_record_schema/base.rb

```diff
@@ -34,6 +41,22 @@ def EntityType(*args)
         @entity_types[entity_type.name] = entity_type
         entity_type
       end
+
+      def transform_json_for_root(json)
```

Comment:
The naming of these seems a bit off. One of them is `transformed` and the others are `transform`. I can see why, but I don't think it's necessary.

Could we have:

```
options[:transformers][:metadata]
options[:transformers][:root]
options[:transformers][:feed]
options[:transformers][:entry]
```

Then you save the `transformers` hash in a single instance variable and expose it publicly. For transformers that are not specified by the user you'd assign a trivial lambda (e.g. `->(x) { x }`, or `-> { self }` for the schema one).

Then out in the code you'd do e.g. `schema.transformers[:root].call(json)`. That way the schema is just holding these lambdas instead of the responsibility for doing all this transformation work directly, which seems like a violation of SRP.

Also I'm not really sure what the purpose of the metadata hook/transformer is. It seems we're not using it anymore given that we're refreshing the schema on each request? It's kind of an order of magnitude more confusing than the other hooks, so if we don't need it then maybe nix it?

Reply:
Agreed with naming suggestions and simplifications!

We do need the metadata transformer actually. The internal data model (schema) is created at request time, but `/$metadata` is a route that returns complex XML with all our repeat groups for PowerBI. This is the one we most need the hook for.